### DISCUSSION TITLE
`plot.variable()` on `predict` object?

### DISCUSSION THREAD

**Discussion by lovestat** [kogalur/randomForestSRC#326] [Q&A]
It seems that the `plot.variable()` only accepts the grown `rfsrc` object, i.e., the trained model and training data. Can I provide another `newdata`/`predict` object to `plot.variable()` to obtain the partial dependence plots? Thanks!!

**Comment by ishwaran** [comment]
`plot.variable` only accepts grow objects because it calls the prediction function in order to calculate partial predicted values.  It creates synthetic data and the predicted value of this synthetic data is determined as a means to obtain partial predicted values.

That is to say, in order to be able to make a prediction call we have to start with a grow object.  You can not start with a prediction object to make prediction calls and there is no way around this.

**Comment by lovestat** [comment]
Thanks a lot for your response, Dr. Ishwaran. I got that making a prediction requires starting with a grown object. 

But what if we'd like to obtain the partial predicted values on another dataset (X_i's) given the model (the prediction function, F_hat). It is like `plot.variable(model, variable, **newdata**)` where the **newdata** can be specified.

**Comment by ishwaran** [comment]
We have discussed this amongst the team, but unfortunately there does not appear to be a way to do this currently.  We will put it on our list of things to do and hopefully get to it sometime soon.

**Reply by lovestat** [reply]
Thanks a lot!!
